realm/realm-library/src/main/java/io/realm/RealmInteger.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm;
+
+
+public final class RealmInteger extends Number implements Comparable<RealmInteger> {
+    private long value;
+
+    /**
+     * Create a new realm integer, with the specified initial value.
+     *
+     * @param value initial value
+     */
+    public RealmInteger(long value) {
+        this.value = value;
+    }
+
+    /**
+     * Create a new realm integer, with the specified initial value.
+     *
+     * @param value initial value: parsed by {@code Long.parseLong}
+     */
+    public RealmInteger(String value) {
+        this.value = Long.parseLong(value);
+    }
+
+    /**
+     * Set the RealmInteger value.
+     * This obliterates any distributed counters and forces the value.
```

Comment:
This description isn't totally clear to me. What about something like: 

"Calling `set()` forcibly sets the RealmInteger to the provided value. Doing this means that {@link increment} and {@link decrement} changes from other devices might be overridden."